src/pip/_internal/operations/freeze.py

```diff
@@ -169,16 +169,9 @@ def _get_editable_info(dist: BaseDistribution) -> _EditableInfo:
     """
     if not dist.editable:
         return _EditableInfo(requirement=None, editable=False, comments=[])
-    if dist.location is None:
-        display = _format_as_name_version(dist)
-        logger.warning("Editable requirement not found on disk: %s", display)
-        return _EditableInfo(
-            requirement=None,
-            editable=True,
-            comments=[f"# Editable install not found ({display})"],
-        )
-
-    location = os.path.normcase(os.path.abspath(dist.location))
+    editable_project_location = dist.editable_project_location
+    assert editable_project_location
+    location = os.path.normcase(os.path.abspath(editable_project_location))
```

Comment:
Should this be

```python
editable_project_location = dist.editable_project_location
if not editable_project_location:
    return _EditableInfo(requirement=None, editable=False, comments=[])
location = os.path.normcase(os.path.abspath(editable_project_location))
```

instead?

Reply:
Yeah, I'm improving that in #10410

Reply:
I can include #10410 here if you prefer but I thought this one was long enough and I can continue improving in followups.

Reply:
It’s OK, let’s do it in #10410.